File: page-objects/js-mocha-chai/complete/pages/article-detail-page.js
var webDriver = require('selenium-webdriver');
var By = webDriver.By;

var BagPage = require('./bag-page.js');

ArticleDetailPage = function ArticleDetailPage(driver) {
    this.driver = driver;
    this.sizeSelect = By.id("sizeSelect");
    this.articlesListLocator = By.className("catalogArticlesList_productBox");
    this.articleBrand = By.css("span[itemprop='brand']");
    this.articleName = By.css("span[itemprop='name']");
    this.addToBagButton = By.id("ajaxAddToCartBtn");
    this.goToBagButton = By.name("head.text:cart.x:4.y:1");
};

ArticleDetailPage.prototype.selectFirstAvailableSize = function() {
    this.driver.findElement(this.sizeSelect).click();
    this.driver.findElements(By.css("li[class='available sizeLine']")).then(function(availableSizes){
        if (availableSizes.length > 0) {
            availableSizes[0].click();
        }
    });
};

ArticleDetailPage.prototype.getArticleName = function() {
     return this.driver.findElement(this.articleName).getText();
};

ArticleDetailPage.prototype.getArticleBrand = function() {
    return this.driver.findElement(this.articleBrand).getText();
};

ArticleDetailPage.prototype.addToBag = function() {
    this.driver.findElement(this.addToBagButton).click();
};

ArticleDetailPage.prototype.goToBag = function() {
    this.driver.findElement(this.goToBagButton).click();
    return new BagPage(this.driver);
};

module.exports = ArticleDetailPage;

File: page-objects/java-testng/complete/src/main/java/pages/ArticleDetailPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.util.List;

public class ArticleDetailPage {

    @SuppressWarnings("FieldCanBeLocal")
    private WebDriver webDriver;

    @FindBy(id = "sizeSelect")
    private WebElement sizeSelect;

    @FindBy(css = "li[class='available sizeLine']")
    private List<WebElement> availableSizes;

    @FindBy(css = "span[itemprop='brand']")
    private WebElement articleBrand;

    @FindBy(css = "span[itemprop='name']")
    private WebElement articleName;

    @FindBy(id = "ajaxAddToCartBtn")
    private WebElement addToBagButton;

    @FindBy(name = "head.text:cart.x:4.y:1")
    private WebElement goToBagButton;

    public ArticleDetailPage(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(this.webDriver, this);
    }

    public void selectFirstAvailableSize() {
        sizeSelect.click();
        availableSizes.get(0).click();
    }

    public String getArticleName() {
        return articleBrand.getText() + " " + articleName.getText();
    }

    public void addToBag() {
        addToBagButton.click();
    }

    public BagPage goToBag() {
        goToBagButton.click();
        return new BagPage(this.webDriver);
    }
}
